### ISSUE TITLE
Is it always necessary to work on branch? When is it alright to work on master?

### ISSUE THREAD

**Issue by fairyinabottle4** [nus-cs2103-AY2021S2/forum#178]
Hi all, I was discussing this question with my teammates when planning the workflow for v1.1. There are some instances where it does not seem like a branch is necessary. One example is updating the "About Us" page, where each of us uploads a picture of ourselves and adds a few other links. Is it necessary to use a branch here? Can we do this on master? Is "branch clutter", i.e. having too many branches, a legit problem?

**Comment by vevek** [comment]
Personally, I feel that pull request are a very good way to review and ensure that code being added to master branch is always in working order. 

Moreover, if another teammate begins working on a new feature from master, they start with a working master and not a potentially buggy one. If he started with a buggy master, it needlessly slows down his development on the feature he wants to work on. 

Lastly, if everyone works on the same branch, I think it would increase the amount of communication between those working on it, as a lot of coordination would be required.

**Comment by damithc** [comment]
In addition to benefits @vevek has mentioned, **one important objective of v1.1 is to practice the forking workflow in a team setting**. So, try to follow the workflow even when you think the same change can be done more easily in another simpler way (and you may be right in most cases).

**Comment by samuelfangjw** [comment]
Regarding your second point on "branch clutter", I believe it is possible to delete a branch after you have merged the changes into master if you don't need it anymore.

**Comment by fairyinabottle4** [comment]
Thank you for your responses. I gather that it is highly highly recommended to work on branches instead of master.

> It is possible to delete a branch after you have merged the changes

@damithc  Can this be done for tp? I recall we were explicitly told not to delete the branches for ip, but it would be great if it were different

**Comment by damithc** [comment]
> @damithc Can this be done for tp? I recall we were explicitly told not to delete the branches for ip, but it would be great if it were different

Good point. Don't delete branches in the tP as our grading scripts look for them when detecting if you have followed the expected workflow. But in other projects, it is common to delete the branch after merging.
